src/rgw/rgw_user.cc

```diff
@@ -2462,8 +2465,11 @@ int RGWUserAdminOp_Subuser::remove(RGWRados *store, RGWUserAdminOpState& op_stat
 
 
   ret = user.subusers.remove(op_state, NULL);
-  if (ret < 0)
+  if (ret < 0) {
+    if (ret == -ENOENT)
+      ret = -ERR_NO_SUCH_SUBUSER;
```

Comment:
@zhaochao You didn't get my point.  I meant: why do you return `ENOENT` and then on the receiving end modify  it to  `ERR_NO_SUCH_SUBUSER` ? You could return the specific error `ERR_NO_SUCH_SUBUSER`, so that it won't conflict with other errors conditions that may return `ENOENT` in the future. Please read my previous comment once again.